speculation-rules-security-privacy-questionnaire.md

```diff
@@ -0,0 +1,105 @@
+
+This is necessary in order to ensure that useful speculated activity can occur to reduce the latency of navigation, while simultaneously limiting adverse effects to the user (e.g. due to limited battery life or metered network bandwidth).
+
+User agents should balance minimization of this information against the benefits of selecting the speculation most likely to be beneficial to the user.
+
+To the extent that the user agent satisfies requirements for client IP anonymization by using a [proxy server](https://github.com/buettner/private-prefetch-proxy), doing so may reveal some information to its operator, such as the destination origin and encrypted network traffic. Feedback on the use of a private prefetch proxy is welcome on a [separate repository](https://github.com/buettner/private-prefetch-proxy/issues).
+
+Much of this exposure is in theory already happening with existing resource hints such as `<link rel=prefetch>`.
+
+### Do features in your specification expose the minimum amount of information necessary to enable their intended uses?
+
+Largely, yes, though the specification leaves some freedom to user agents to determine whether to perform a prefetch, prerender or similar, and their implementation choices may determine whether more than the minimum amount of information is exposed through side channels.
+
+### How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?
+
+This is substantially similar to the answer to the first question.
+
+The prefetches/prerenders themselves may include PII if a site includes them in the requested URL (as with any outgoing fetch or navigation).
+
+The fact that a prefetch or prerender occurs is visible to its origin server. Accordingly, user agents should avoid basing this decision on personal information not already known to the site. For example, if a user agent prefetched a link based on a profile of the user's interests from their browsing activity across the web, the fact that a prefetch was executed may reveal information about those interests that would not have otherwise been exposed.
+
+To the extent that the user agent modifies the path taken by network traffic in order to accommodate a request to anonymize the client IP address, encrypted traffic that may be derived from PII may be visible to parties along the revised network path, including the proxy operator To mitigate this, traffic should be encrypted with TLS and not include cookies or other identifiers.
```

Comment:
```suggestion
To the extent that the user agent modifies the path taken by network traffic in order to accommodate a request to anonymize the client IP address, encrypted traffic that may be derived from PII may be visible to parties along the revised network path, including the proxy operator. To mitigate this, traffic should be encrypted with TLS and not include cookies or other identifiers.
```

Reply:
"...as is [required](https://github.com/jeremyroman/alternate-loading-modes/blob/main/fetch.md#fetching-with-no-credentials) by other parts of the proposal" maybe?

Reply:
Compromised and used that link on the phrase "not include cookies or other identifiers" in the interest of brevity; hopefully that captures it.